pkg/services/publicdashboards/commands/generate_datasources/generate.go

```diff
@@ -0,0 +1,105 @@
+package generate_datasources
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"sync"
+)
+
+type datasourceMeta struct {
+	Slug string `json:"slug"`
+}
+
+type listPluginResponse struct {
```

Comment:
Given that a bunch of these types are created just for unmarshaling, I think it makes sense to do it inline instead of defining a bunch of extra types. Might need to check syntax, but we should be able to do this below as well.

```suggestion
type listPluginResponse struct {
	Items []{
	  Slug string `json:"slug"`
	 } `json:"items"`
}
```

Reply:
nice, I'll try and do that all in one go